korman/nodes/node_avatar.py

```diff
@@ -71,3 +71,388 @@ def export(self, exporter, bo, so):
+        }),
+    ])
+
+    output_sockets = OrderedDict([
+        ("stage", {
+            "text": "Behavior",
+            "type": "PlasmaAnimStageRefSocket",
+            "valid_link_nodes": "PlasmaMultiStageBehaviorNode",
+            "valid_link_sockets": "PlasmaAnimStageRefSocket",
+        }),
+        ("stage_reference", {
+            "text": "Stage Progression",
+            "type": "PlasmaAnimStageOrderSocketOut",
+            "valid_link_nodes": "PlasmaAnimStageSettingsNode",
+            "valid_link_sockets": {"PlasmaAnimStageAdvanceSocketIn", "PlasmaAnimStageRegressSocketIn"} ,
+        }),
+    ])
+
+    def draw_buttons(self, context, layout):
+        col = layout.column()
+        col.prop(self, "anim_name")
+        col.separator()
+
+        row = col.row()
+        row.prop(self, "loop_option")
+        if self.loop_option == "kLoop":
+            row = col.row()
+            row.prop(self, "num_loops")
+        col.separator()
+
+    @property
+    def stage_id(self):
+        idx = None
+        if self.find_output_socket("stage").is_linked:
+            parent_MSB = self.find_output_socket("stage").links[0].to_node
+            idx = next((idx for idx, socket in enumerate(parent_MSB.find_input_sockets("stage_refs")) if socket.is_linked and socket.links[0].from_node == self))
+        return idx
+
+
+class PlasmaBehaviorSocket(PlasmaNodeSocketBase, bpy.types.NodeSocket):
+    bl_color = (0.348, 0.186, 0.349, 1.0)
+
+
+class PlasmaMultiStageBehaviorNode(PlasmaNodeBase, bpy.types.Node):
+    bl_category = "AVATAR"
+    bl_idname = "PlasmaMultiStageBehaviorNode"
+    bl_label = "Multistage Behavior"
+    bl_width_default = 200
+
+    pl_attrib = ("ptAttribBehavior")
+
+    freeze_phys = BoolProperty(name="Freeze Physical",
+                              description="Freeze physical at end",
+                              default=False)
+    reverse_control = BoolProperty(name="Reverse Controls",
+                              description="Reverse forward/back controls at end",
+                              default=False)
+
+    input_sockets = OrderedDict([
+        ("seek_target", {
+            "text": "Seek Target",
+            "type": "PlasmaSeekTargetSocketIn",
+            "valid_link_sockets": {"PlasmaSeekTargetSocketOut"},
+        }),
+        ("stage_refs", {
+            "text": "Stage",
+            "type": "PlasmaAnimStageRefSocket",
+            "valid_link_nodes": "PlasmaAnimStageNode",
+            "valid_link_sockets": "PlasmaAnimStageRefSocket",
+            "link_limit": 1,
+            "spawn_empty": True,
+        }),
+    ])
+
+    output_sockets = OrderedDict([
+        ("hosts", {
+            "text": "Host Script",
+            "type": "PlasmaBehaviorSocket",
+            "valid_link_nodes": {"PlasmaPythonFileNode"},
+            "spawn_empty": True,
+        })
+    ])
+
+    def draw_buttons(self, context, layout):
+        col = layout.column()
+        col.label("Options:")
```

Comment:
The other nodes that have options like these don't have this label :wink:

Reply:
Good catch.  I think this was a leftover from the early mockup stage.  💥